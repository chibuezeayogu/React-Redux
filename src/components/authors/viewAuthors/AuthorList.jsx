import React, { Component } from 'react';
import { object, func } from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * @class AuthorList
 * @extends Compoent
 * 
 */
class AuthorList extends Component {
/**
 * @memberof AuthorList
 * @method handleDeleteCourse
 * @param {string} id
 * @returns {void}
 */
	handleDeleteAuthor = (id) => {
		this.props.deleteAuthor(id);
	}
	
	render() {
		const { author } = this.props;
		return (
			<tr>
				<td><Link to={`/author/${author.id}`}>{author.id}</Link></td>
				<td>{author.firstName}</td>
				<td>{author.lastName}</td>
				<td>
					<button 
						type="button" 
						onClick={() => this.handleDeleteAuthor(author.id)} 
						className="btn btn-danger"
					>
							Delete
					</button>
				</td>
			</tr>
		);
	}
}

AuthorList.propTypes = {
	course: object,
	showDeleteModal: func
};

export default AuthorList;
