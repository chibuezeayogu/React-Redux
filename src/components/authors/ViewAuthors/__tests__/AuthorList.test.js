import React from 'react';
import { shallow } from 'enzyme';

import AuthorList from '../AuthorList';

describe('CourseList Component', () => {
	const props = {
		deleteAuthor: jest.fn(),
		author: {
			id: 'chibueze-ayogu',
			firstName: 'Chibueze',
			lastName: 'Ayugo'
		}
	};

	const wrapper = shallow(<AuthorList {...props} />);

	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('button')).toHaveLength(1);
		expect(wrapper.find('button').text()).toEqual('Delete');
		expect(wrapper.find('tr')).toHaveLength(1);
	});

	it('should call handleDeleteAuthor method', () => {
		const spy = jest.spyOn(wrapper.instance(), 'handleDeleteAuthor');
		wrapper.instance().handleDeleteAuthor({ id: 'chibueze-ayogu' });
		expect(spy).toHaveBeenCalled();
	});

	it('should simulate delete button click', () => {
		expect(wrapper.find('button')).toHaveLength(1);
		wrapper.find('button').simulate('click');
	});
});
