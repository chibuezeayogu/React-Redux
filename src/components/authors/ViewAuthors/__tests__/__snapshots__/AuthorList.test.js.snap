// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList Component should render correctly 1`] = `
<tr>
  <td>
    <Link
      replace={false}
      to="/author/chibueze-ayogu"
    >
      chibueze-ayogu
    </Link>
  </td>
  <td>
    Chibueze
  </td>
  <td>
    Ayugo
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
